runtime/src/bank.rs

```diff
@@ -3651,6 +3653,39 @@ impl Bank {
         }
     }
 
+    fn simulation_override_accounts(&self) -> AccountOverrides {
+        let mut account_overrides = AccountOverrides::new();
+        self.add_account_override(
```

Comment:
We only need to add the override when `slot_history.check(self.slot()) == Check::Found`. Simulation can currently only be done on a frozen bank but maybe that could change in the future.

Reply:
Maybe I'm missing something, but slot history also encodes the slot, which is the origin of this exploit.  Why do you we only need to override if the slot is found?